src/Sylius/Bundle/ApiBundle/CommandHandler/PickupCartHandler.php

```diff
@@ -71,7 +68,7 @@ public function __invoke(PickupCart $pickupCart)
         /** @var ChannelInterface $channel */
         $channel = $this->channelRepository->findOneByCode($pickupCart->getChannelCode());
 
-        $customer = $this->provideCustomer();
+        $customer = $this->customerProvider->get();
```

Comment:
The same case as with adding a product review, the user shouldn't be provided in handler from context, I would be for reverting this refactor in this PR and do it in a separate PR